### PR TITLE
Fetch theme editor items from ThemeDB

### LINKED ISSUE #82402: Theme items that don't exist in the default theme are not listed in the editor
### Godot version

4.2 251fb83

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

Button has `icon` theme item:
image
But it does not appear in the overrides:
image
Nor in the theme editor:
image

### Steps to reproduce

1. Add Button
2. Try to find `icon` theme property (not the Button's own property)
3. It exists only in theory (documentation)

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by KoBeWi**
Together with #84760 fixes #82402

### CODE REVIEW

File: editor/plugins/theme_editor_plugin.cpp [hunk 1]
```diff
@@ -2370,18 +2370,25 @@ void ThemeTypeEditor::_update_type_list_debounced() {
...
 		}
 
-		(ThemeDB::get_singleton()->get_default_theme().operator->()->*get_list_func)(default_type, &names);
+		List<ThemeDB::ThemeItemBind> theme_binds;
+		ThemeDB::get_singleton()->get_class_own_items(default_type, &theme_binds);
+		for (const ThemeDB::ThemeItemBind &E : theme_binds) {
+			if (E.data_type == p_type) {
+				names.push_back(E.item_name);
+			}
+		}
```
Comment (YuriSizov) [inline editor/plugins/theme_editor_plugin.cpp:2390]: It's probably worth it to add a new method to filter by the data type and avoid excessive loops.

**Review by YuriSizov — APPROVED** [review]
Needs a rebase, but otherwise looks good.

Do you think we leave any performance on the table by switching from a method pointer to `get_theme_item_list`? I know this is needed for `ThemeDB` anyway, so it's fine. But I'm not well versed in C++ so curious what your take is here.

File: editor/plugins/theme_editor_plugin.cpp (hunk 1 above)
Comment (YuriSizov) [reply editor/plugins/theme_editor_plugin.cpp:2390]: Looks like it's needed in https://github.com/godotengine/godot/pull/85015 as well.

**Review by YuriSizov — CHANGES_REQUESTED** [review]
Actually, as I was testing https://github.com/godotengine/godot/pull/84760, I realized there is something wrong will all 3 PRs. While we can use `get_class_own_items` in the docs (and in fact we should), for the theme editor, the override list, and the argument autocompletion we need to include inherited items as well.

This requires a new `get_class_items` method, which returns not just own items. Plus we need a method to get such items filtered based on the data type, as outlined above. I think one of these PRs need to do all this groundwork, and then others can be built on top of it.

Let me know if you need any help with any of that.

PS. If you need an example of this being a problem, check CheckBox for instance.

**Comment by KoBeWi** [comment]
Updated. I added yet another argument to `get_class_items()`, which can be used to filter by type. If I were to add a new method, it would be pretty much the same, so I wanted to avoid code duplication.

**Review by akien-mga — APPROVED** [review]
Code changes look fine to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**